README.md

```diff
@@ -171,6 +171,7 @@ For more information on the structure of the YAML configuration file, please vis
 | `containerAppEnvironment` | No       | The name of the Container App environment to use with the application. If not provided, an existing environment in the resource group of the Container App will be used, otherwise, an environment will be created in the formation `<container-app-name>-env`. |
 | `runtimeStack`            | No       | The platform version stack used in the final runnable application image that is deployed to the Container App. The value should be provided in the formation `<platform>:<version>`. If not provided, this value is determined by Oryx based on the contents of the provided application. Please refer to [this document](https://github.com/microsoft/Oryx/blob/main/doc/supportedRuntimeVersions.md) for more information on supported runtime stacks for Oryx. |
 | `builderStack`            | No       | The stack (OS) that should be used to build the provided application source and produce the runnable application image. You can provide a specific image tag for the stack, such as "debian-bookworm-20231107.2", or you can provide a supported stack name, such as "debian-bookworm" or "debian-bullseye", and the latest supported image tag for that stack will be used. If no stack is provided, this action will attempt to build the provided application source with each supported stack until there's a successful build. |
+| `buildEnvironmentVariables` | No     | A list of environment variable(s) used to configure the Buildpacks for image build. Space-separated values in 'key=value' format. Empty string to clear existing values. |

```

Comment:
We previously introduced a [`buildArguments`](https://github.com/Azure/container-apps-deploy-action/pull/68/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R91-R97) input for this GitHub Action that would allow customers to provide custom environment variables for (1) their Dockerfile in the `docker build` flow and (2) the buildpack environment when running `pack build`.

We ended up reverting this change when it wasn't entirely clear how this would be used in the source-to-cloud flow when it was introduced in the CLI and took precedence over the local `pack build` we were running within this action.

I think in this case, it would be a good idea to bring back this `buildArguments` input to specifically handle these two cases, and later on propagate the value(s) to the CLI when the commands can handle these environment variables being provided and will automatically propagate them to the cloud build.

Reply:
1. When @daniv-msft and I chatted previously, I believe we settled on `buildArguments` since it made sense from a `docker build` and `pack build` standpoint where these arguments would be used (_i.e._, `--build-arg` for `docker build` and arguments for the "build environment" for `pack build`). Do you have any issues with making this input called `buildArguments`?

2. In the previous PR, we just used a comma to separate the values, but I don't think we had any strong input on that-- if the Azure CLI is using spaces and that's how you have it currently implemented, then I'm OK with keeping that format.

Reply:
#1: Let's use `buildArguments` as you guys have made decision before.
#2: Let's use spaces to be consistent with the CLI, which may be easier for user to understand.